src/DSharp.Compiler/Compiler/ExpressionBuilder.cs

```diff
@@ -1795,6 +1797,75 @@ private Expression ProcessOpenParenExpressionNode(BinaryExpressionNode node)
             return methodExpression;
         }
 
+        private (ExpressionListNode argumentNodes, List<Expression> argumentExpressions) ParseArguments(BinaryExpressionNode node, MemberExpression memberExpression)
+        {
+            ExpressionListNode argumentNodes = null;
+            List<Expression> argumentExpressions = new List<Expression>();
+
+            if (node.RightChild == null)
+            {
+                return (argumentNodes, argumentExpressions);
+            }
+
+            Debug.Assert(node.RightChild is ExpressionListNode);
+
```

Comment:
maybe we should throw a real exception with the details of what is throwing an exception

Reply:
we have access to ErrorHandler in the ExpressionBuilder - do we have enough context to give a meaningful build error here?

Reply:
I've updated this to throw an exception with more information